src/filemanager.cpp

```diff
@@ -1203,12 +1203,12 @@ const char *proj_context_get_user_writable_directory(PJ_CONTEXT *ctx,
     if (ctx->user_writable_directory.empty()) {
         std::string path;
 #ifdef _WIN32
-        std::wstring wPath;
-        wPath.resize(MAX_PATH);
-        if (SHGetFolderPathW(nullptr, CSIDL_LOCAL_APPDATA, nullptr, 0,
-                             &wPath[0]) == S_OK) {
-            wPath.resize(wcslen(wPath.data()));
-            path = NS_PROJ::WStringToUTF8(wPath);
+      wchar_t* wPath;
+        if (SHGetKnownFolderPath(FOLDERID_LocalAppData, 0, nullptr, &wPath) == S_OK){
```

Comment:
to avoid the mingw32 build failure, we could perhaps keep the old detection code and use if only under #ifdef __MINGW32__ ?